media/starboard/starboard_cdm.cc

```diff
@@ -389,190 +342,205 @@ void DrmSystem::OnSessionUpdated(int ticket, SbDrmStatus status,
 
   // Interpret the result.
   if (status == kSbDrmStatusSuccess) {
-    session_update.updated_callback.Run();
-  } else {
-    session_update.did_not_update_callback.Run(status, error_message);
+    promises_.ResolvePromise(session_update.promise_id);
```

Comment:
Do we need to reject the promise if `status` isn't `kSbDrmStatusSuccess`?

Reply:
done